### PR TITLE
redid explosives system; fixed shack demolition

### PR DESCRIPTION
**Pull request by Z1Meme**

### CODE REVIEW

**Comment by Z1Meme** [comment]
Needs to be heavily tested before merging.

**Comment by Z1Meme** [comment]
@QuentinGruber I fully tested this and it should be ready

File: src/servers/ZoneServer2016/entities/constructiondoor.ts [hunk 1]
```diff
@@ -73,7 +94,7 @@ export class ConstructionDoor extends DoorEntity {
     this.slot = slot;
     this.profileId = 999; /// mark as construction
```
Comment (QuentinGruber) [inline src/servers/ZoneServer2016/entities/constructiondoor.ts:95]: custom profileIds should be a constant

Comment (QuentinGruber) [reply src/servers/ZoneServer2016/entities/constructiondoor.ts:95]: i know it's not from that pr, it's just a reminder so i will create an issue for it

File: src/servers/ZoneServer2016/zoneserver.ts [hunk 2]
```diff
@@ -2220,6 +2220,8 @@ export class ZoneServer2016 extends EventEmitter {
       }
     }
 
+    const baseConstructionDamage = 34000;
```
Comment (QuentinGruber) [inline src/servers/ZoneServer2016/zoneserver.ts:2223]: BaseConstructionDamage could be a configurable var

**Review by QuentinGruber — APPROVED** [review]

**Pull request merged**